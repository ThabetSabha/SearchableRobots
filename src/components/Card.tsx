import React from "react";

interface CardProps {
  id: number;
  name: string;
  email: string;
}

const Card = ({ id, name, email }: CardProps) => {
  return (
    <div
      className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
      style={{ minWidth: "350px" }}
    >
      <img alt="robots" src={`https://robohash.org/${id}?size=200x200`}></img>
      <div>
        {" "}
        <h2> {name} </h2>
        <p> {email} </p>
      </div>
    </div>
  );
};

export default Card;
