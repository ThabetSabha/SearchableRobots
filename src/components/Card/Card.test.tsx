import React from "react";
import Card from "./Card";
import { shallow } from "enzyme";

test("That the Card Component Renders", () => {
  let cardProps: CardProps = {
    id: 1,
    name: "Test",
    email: "test@gmail.com",
  };

  let wrapper = shallow(<Card {...cardProps} />);
  expect(wrapper).toMatchSnapshot();
});
