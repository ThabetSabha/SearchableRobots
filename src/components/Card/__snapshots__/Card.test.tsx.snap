// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`That the Card Component Renders 1`] = `
<div
  className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
  style={
    Object {
      "minWidth": "350px",
    }
  }
>
  <img
    alt="robots"
    src="https://robohash.org/1?size=200x200"
  />
  <div>
     
    <h2>
       
      Test
       
    </h2>
    <p>
       
      test@gmail.com
       
    </p>
  </div>
</div>
`;
